Add deploy  .sh script

rochacbruno opened the pull request:
Experimenting with a script to install everything.

Tested in Fedora 24 running on Qemu/kvm

rochacbruno committed:
Add deploy_my_machine script

rochacbruno committed:
fix some package names

elyezer commented on dev/playbooks/README.md:
````diff
@@ -1,11 +1,22 @@
+### auto
+
+```bash
+wget -O /tmp/deploy_my_machine.sh https://raw.githubusercontent.com/SatelliteQE/sat6helper/master/dev/playbooks/deploy_my_machine.sh && . /tmp/deploy_my_machine.sh
````
You can make this one liner install to not require writing to a local file like this https://gist.github.com/elyezer/d1b011f1f5ac66b35b0f#gistcomment-1725944

elyezer commented on dev/playbooks/deploy_my_machine.sh:
```diff
@@ -0,0 +1,113 @@
+# http://stackoverflow.com/a/2654048/2975300
+
+# http://stackoverflow.com/a/28393320/2975300
+
```
I would recommend defining `set -euo pipefail` which is the unofficial bash strict mode http://redsymbol.net/articles/unofficial-bash-strict-mode/

This will help debugging as well as help stopping the script if something fails.

elyezer commented on dev/playbooks/roles/fedora-common-packages/tasks/main.yml:
```diff
@@ -1,5 +1,5 @@
 - name: update packages cache
-  yum: update_cache=yes
+  yum: name=yum update_cache=yes
```
You can do something like `action: "{{ ansible_pkg_mgr }} update_cache=yes"`. That will call yum or dnf depending on the available command, on Fedora 23 and 24 that should run just DNF.

elyezer commented:
ACK, left some non-blocking comments.

sthirugn commented:
I am getting this error, I forked this in my f24 vm and ran it locally:

``` sh
$ /bin/bash dev/playbooks/deploy_my_machine.sh 

Starting Installation
[sudo] password for suresh:
[sudo] password for suresh: Last metadata expiration check: 0:23:20 ago on Tue Jul  5 09:41:04 2016.
Dependencies resolved.
Nothing to do.
Complete!
Last metadata expiration check: 0:23:21 ago on Tue Jul  5 09:41:04 2016.
Package wget-1.18-1.fc24.x86_64 is already installed, skipping.
Package unzip-6.0-29.fc24.x86_64 is already installed, skipping.
Package python-2.7.11-6.fc24.x86_64 is already installed, skipping.
Package python-devel-2.7.11-6.fc24.x86_64 is already installed, skipping.
Package libffi-devel-3.1-9.fc24.x86_64 is already installed, skipping.
Package redhat-rpm-config-40-2.fc24.noarch is already installed, skipping.
Package openssl-devel-1:1.0.2h-1.fc24.x86_64 is already installed, skipping.
Package yum-3.4.3-509.fc24.noarch is already installed, skipping.
Dependencies resolved.
Nothing to do.
Complete!
--2016-07-05 10:04:27--  https://github.com/SatelliteQE/sat6helper/archive/master.zip
Resolving github.com (github.com)... 192.30.253.113
Connecting to github.com (github.com)|192.30.253.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://codeload.github.com/SatelliteQE/sat6helper/zip/master [following]
--2016-07-05 10:04:27--  https://codeload.github.com/SatelliteQE/sat6helper/zip/master
Resolving codeload.github.com (codeload.github.com)... 192.30.253.120
Connecting to codeload.github.com (codeload.github.com)|192.30.253.120|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [application/zip]
Saving to: ‘/tmp/deploy_my_machine/sat6helper.zip’

/tmp/deploy_my_machine/s     [ <=>                              ]   7.11K  --.-KB/s    in 0.002s  

2016-07-05 10:04:27 (4.33 MB/s) - ‘/tmp/deploy_my_machine/sat6helper.zip’ saved [7280]

Archive:  sat6helper.zip
fb7909af6331315905154669b914bfeb0d507b72
   creating: sat6helper-master/
  inflating: sat6helper-master/README.md  
   creating: sat6helper-master/dev/
   creating: sat6helper-master/dev/playbooks/
  inflating: sat6helper-master/dev/playbooks/README.md  
  inflating: sat6helper-master/dev/playbooks/ansible.cfg  
  inflating: sat6helper-master/dev/playbooks/default_fedora_packages_to_install.yml  
  inflating: sat6helper-master/dev/playbooks/hosts  
   creating: sat6helper-master/dev/playbooks/roles/
   creating: sat6helper-master/dev/playbooks/roles/fedora-common-packages/
   creating: sat6helper-master/dev/playbooks/roles/fedora-common-packages/tasks/
  inflating: sat6helper-master/dev/playbooks/roles/fedora-common-packages/tasks/main.yml  
   creating: sat6helper-master/dev/playbooks/roles/git/
   creating: sat6helper-master/dev/playbooks/roles/git/tasks/
  inflating: sat6helper-master/dev/playbooks/roles/git/tasks/main.yml  
   creating: sat6helper-master/dev/playbooks/roles/git/templates/
  inflating: sat6helper-master/dev/playbooks/roles/git/templates/gitconfig.j2  
   creating: sat6helper-master/dev/playbooks/roles/pycharm/
   creating: sat6helper-master/dev/playbooks/roles/pycharm/defaults/
  inflating: sat6helper-master/dev/playbooks/roles/pycharm/defaults/main.yml  
   creating: sat6helper-master/dev/playbooks/roles/pycharm/tasks/
  inflating: sat6helper-master/dev/playbooks/roles/pycharm/tasks/main.yml  
   creating: sat6helper-master/dev/playbooks/roles/pycharm/templates/
  inflating: sat6helper-master/dev/playbooks/roles/pycharm/templates/jetbrains-pycharm.desktop.j2  
  inflating: sat6helper-master/dev/playbooks/workstation.yml  
running playbook

PLAY [home] ********************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [fedora-common-packages : update packages cache] **************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "one of the following is required: name,list"}

NO MORE HOSTS LEFT *************************************************************
    to retry, use: --limit @workstation.retry

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=1   
```

rochacbruno commented:
@sthirugn we need to merge first, because the download is getting form master.

sthirugn commented:
@rochacbruno I didnt mention earlier.  I ran it from your branch directly.

sthirugn commented:
@rochacbruno ok then lets merge and test it out - ok for me.

ACK

rochacbruno commented:
@sthirugn let me address some of @elyezer comments

rochacbruno commented:
``` bash
[bruno@localhost playbooks]$ ansible-playbook workstation.yml --extra-vars "ansible_become_pass=secret" 

PLAY [home] ********************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [fedora-common-packages : update packages cache] **************************
changed: [localhost]
 [WARNING]: Consider using dnf module rather than running dnf


TASK [fedora-common-packages : upgrade all packages] ***************************
ok: [localhost]

TASK [fedora-common-packages : ensure package list is installed] ***************
ok: [localhost] => (item=[u'@Development Libraries', u'aria2', u'automake', u'gcc', u'gcc-c++', u'kernel-devel', u'diffstat', u'doxygen', u'patch', u'patchutils', u'colordiff', u'meld', u'qgit', u'cheese', u'cmake', u'curl', u'gettext', u'gimp', u'gparted', u'graphviz-devel', u'htop', u'ImageMagick', u'inkscape', u'libmemcached-devel', u'libwebp-devel', u'lynx', u'macchanger', u'p7zip', u'python3-devel', u'python-devel', u'unetbootin', u'wget', u'xclip', u'xml2', u'zlib-devel', u'libffi-devel', u'openssl-devel', u'pidgin', u'docker', u'libselinux-python', u'python-dnf'])

TASK [git : ensure git is installed] *******************************************
ok: [localhost]

TASK [git : be sure git is configured] *****************************************
ok: [localhost]

TASK [pycharm : ensure if /home/bruno/binaries/pycharm directory exists] *******
ok: [localhost]

TASK [pycharm : ensure aria2 download manager is installed] ********************
ok: [localhost]

TASK [pycharm : download Pycharm (280MB)] **************************************
changed: [localhost]

TASK [pycharm : ensure pycharm is installed] ***********************************
skipping: [localhost]

TASK [pycharm : ensure if /home/bruno/.local/share/applications directory exists] ***
ok: [localhost]

TASK [pycharm : ensure pycharm launcher is installed] **************************
skipping: [localhost]

PLAY RECAP *********************************************************************
localhost                  : ok=10   changed=2    unreachable=0    failed=0  

```

rochacbruno commented:
@elyezer the one liner install thing will be implemented later :) thanks for suggestion

rochacbruno committed:
Using package instead of explicit using yum or dnf

rochacbruno committed:
added tags so we can choose which roles to run

rochacbruno commented:
Just tested in Fedora 24 clean VM, worked, we can merge this one, then we can move to implement new tasks. and also added tags so we use `--tags pycham` to choose roles

omaciel commented:
@rochacbruno is installing `pycharm` optional? If not, I think it should...

rochacbruno commented:
@omaciel I included tags, so the way to run should be with `--skip-tags pycham,emacs,vlc`
